### DISCUSSION TITLE
Read and Write Compatibility Issues

### DISCUSSION THREAD

**Discussion by simonvincenthk** [boisgera/pandoc#62] [Q&A]
Hi,

I'd like to use this `pandoc` wrapper to convert complex `*.docx` reports into `*.md` (markdown). The fact that this wrapper preserves `pandoc`'s (the core haskel library's) modular structure is helpful. Successful conversion will require I alter the AST after reading and before writing. (The other commonly used wrapper, `pypandoc`, does not preserve the modular structure and, therefore, would not allow me to alter the AST...) 

I've run up against a few issues using this wrapper, through:

(1) **When writing to markdown, there is a version incompatibility issue.**

```
import pandoc
text = "Hello world!"
doc = pandoc.read(text)
md = pandoc.write(doc, file = 'HelloWorld.md', format = 'markdown')
```
```
>> JSON parse error: Error in $: Incompatible API versions: encoded with [1,22,2,1] but attempted to decode with [1,23,1].
```
(2) **When configuring the wrapper to write with version 1.22.2.1, the writer does not recognize the output of the reader as a `Pandoc` object.**
```
import pandoc
text = "Hello world!"
pandoc.configure(reset=True)
doc = pandoc.read(text)
pandoc.configure(pandoc_types_version='1.22.2.1') 
md = pandoc.write(doc, file = 'HelloWorld.md', format = 'markdown')
```
```
>> TypeError: Pandoc(Meta({}), [Para([Str('Hello'), Space(), Str('world!')])]) is not a Pandoc, Block or Inline instance.
```

Help is appreciated!

Best regards, Simon

**Comment by simonvincenthk** [comment]
I've managed to temporarily resolve the read and write incompatibility issues arising from pandoc `version` and `pandoc_types_version`  by using the last supported version of pandoc ([2.19.2](https://github.com/jgm/pandoc/releases/tag/2.19.2)) and [configuring pandoc automatically](https://boisgera.github.io/pandoc/api/).

Checking pandoc's version in the CLI:

```
pandoc --version
```
```
>> pandoc.exe 2.19.2
```

Automaticaly cofiguring pandoc:

```
pandoc.configure(auto = True)
```

**Reply by boisgera** [reply]
Hi @simonvincenthk !

Thanks for your feedback. Could you tell me what version of pandoc you were using initially?

The current version of pandoc-the-python-library is tested against pandoc-the-Haskell-CLI 3.1.1 so you *should* be ok up to this version.

Note that if you import pandoc.types, the automatic configuration happens automatically if no manual configuration was done beforehand.

Regards,

Sébastien
